### PR TITLE
libschedutil: export library for use by flux-sched and others

### LINKED ISSUE #2377: export schedutil library for use by flux-sched (and others)
### TODO

- [x] install the compiled shared library
- [x] install the headers
- [x] create a .pc file
- [x] prefix all public function typedefs with `schedutil_`
- [ ] ~~create a man page~~

### Question

Right now the public schedutil header is actually a composite of five different headers (`hello.h`, `ready.h`, `alloc.h`, `free.h`, `ops.h`).  To avoid naming conflicts, do we want to:

- merge them into a single `schedutil.h` header?
- install them into a `schedutil` directory within `$PREFIX/include` or `$PREFIX/include/core`?
- prefix them all with `schedutil_`?

Since the individual headers are relatively small, I'm leaning towards just merging them, but I can also see why you wouldn't want to lose the structure/separation.  Any thoughts?

### PR DESCRIPTION
**Pull request by SteVwonder**
- installs the compiled schedutil shared library
- adds a libschedutil map file for `ld`
- installs the schedutil headers
- creates a schedutil .pc file
- prefixes all public function typedefs with `schedutil_`

Closes #2377

### CODE REVIEW

File: etc/flux-schedutil.pc.in [hunk 1]
```diff
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: flux-idset
```
[the remark this answers is not in the document]
Comment (SteVwonder) [reply etc/flux-schedutil.pc.in]: Nice catch! I would've missed that one. Fixed in the latest force-push.

**Comment by SteVwonder** [comment]
Just finished the corresponding flux-sched PR on my local machine and everything seems to work just fine.  Thanks @grondo for the review.  Once Travis goes green, I'm happy with this PR.

**Comment by SteVwonder** [comment]
The clang builder failed with:
```
+ flux job attach 333514276864

+ flux job info 333514276864 eventlog jobspec R

==30549== 

==30549== HEAP SUMMARY:

==30549==     in use at exit: 22,719 bytes in 58 blocks

==30549==   total heap usage: 80,679 allocs, 80,621 frees, 56,210,722 bytes allocated

==30549== 

==30549== LEAK SUMMARY:

==30549==    definitely lost: 0 bytes in 0 blocks

==30549==    indirectly lost: 0 bytes in 0 blocks

==30549==      possibly lost: 0 bytes in 0 blocks

==30549==    still reachable: 22,719 bytes in 58 blocks

==30549==         suppressed: 0 bytes in 0 blocks

==30549== Reachable blocks (those to which a pointer was found) are not shown.

==30549== To see them, rerun with: --leak-check=full --show-leak-kinds=all

==30549== 

==30549== For counts of detected and suppressed errors, rerun with: -v

==30549== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)

flux-start: 0 (pid 30548) Killed

not ok 1 - valgrind reports no new errors on 2 broker run
```

I'm a little suspicious that a different compiler would result in a leak.  I'm guessing something hung and the test timed out?  I'll try restarting it and maybe it'll pass.

**Comment by garlick** [comment]
LGTM!  Thanks!

**Comment by SteVwonder** [comment]
Thanks @garlick!

**Pull request merged**